src/olympia/shelves/tests/test_views.py

```diff
@@ -55,14 +53,34 @@ def test_basic(self):
             'previous': None,
             'results': []}
 
-        hpshelf_a.update(enabled=True)
-        hpshelf_b.update(enabled=True)
-        # don't enable the last homepage shelf
+    def test_only_enabled_shelves_in_view(self):
+        self.hpshelf_a.update(enabled=True)
+        self.hpshelf_b.update(enabled=True)
+        # don't enable shelf_c
+
         with self.assertNumQueries(4):
             response = self.client.get(self.url)
         assert response.status_code == 200
-        result = json.loads(response.content)
 
+        result = json.loads(response.content)
         assert len(result['results']) == 2
-        assert result['results'][0]['title'] == 'Enhanced privacy extensions'
-        assert result['results'][1]['title'] == 'Recommended extensions'
+        assert result['results'] == [
+            {
+                'title': 'Enhanced privacy extensions',
+                'url': self.collections_url,
+                'endpoint': 'collections',
+                'criteria': 'privacy-matters',
+                'footer_text': 'See more enhanced privacy extensions',
+                'footer_pathname': '',
+                'addons': None
+            },
+            {
+                'title': 'Recommended extensions',
+                'url': self.search_url,
+                'endpoint': 'search',
+                'criteria': '?promoted=recommended&sort=random&type=extension',
+                'footer_text': 'See more recommended extensions',
+                'footer_pathname': '',
+                'addons': []
+            },
+        ]
```

Comment:
Can we have a test that includes at least one addon to test the complete output?  

As `AddonSearchView` calls elasticsearch the `TestCase` will need to change to `ESTestCase` - which is a little different.   Have a look through the tests that use it, but main difference is the addon setup is in `setUpTestData` (a `@classmethod`) and you have to do `cls.refresh()` after all the setup is done to trigger an elasticsearch index update.